app/create.js

```diff
@@ -17,74 +17,113 @@ define(['app/config', 'app/database', 'jquery', 'bootstrap', 'parsley', 'app/sel
 			numberOfFormItems--;
 		}
 	}
-  
-	// Function for adding elements to the form
-	function add_element() {
-		var newElement = $('#item-layout').clone(true).appendTo("#items table").removeAttr("id");
-		newElement.html(giveId(newElement.html(), formItemId));
-    newElement.html(giveRequired(newElement.html()));
+
+	function add_element_to_form() {
+		var newElement = $('#items table').cloneLayout();
+		newElement.replaceClone(["i", "required"], [formItemId, "required"]);
+		// Remove when clicked on close
 		newElement.on("click", ".remove", function() {
 			remove_element($(this));
 		});
+		// When the TAB is pressed, add a new line
+		removeKeybinds("keydown");
+		newElement.find("input:last").on('keydown', function(e) {
+			if (isTab(e.keyCode)) {
+					add_element_to_form();
+			}
+		});
 		numberOfFormItems++;
 		formItemId++;
 	}
-  
-  // Auxiliary replace functions
-  function giveId(string, formItemId) {
-    return string.replace(/{i}/g, formItemId);
-  }
-  function giveRequired(string) {
-    return string.replace(/{required}/g, 'required=""');
-  }
-  
-  // Auxiliary form functions
+
+	function isTab(keyCode){
+		return keyCode == config.key("TAB");
+	}
+
+	function removeKeybinds(keybind) {
+		$("#items table input").each(function() {
+			$(this).unbind(keybind);
+		});
+	}
+
+	// Auxiliary form functions
   function getItemVal(formName, formIndex) {
     return $("#items input[name='" + formName + formIndex + "']").val();
   }
-  
-	// Add the first element
-	add_element();
+	function getFormVal(parentName, formType, formName) {
+    return $(parentName).find(formType + '[name="' + formName + '"]').val();
+  }
 
-	$(".add").click(function(add) {
-		add_element();
-		return false;
-	});
+	// Function for showing the user the system is loading
+	function showLoading(onSuccess) {
+		$("#loadFrame").children("h1").html(lang("create_busycreating"));
+		$("#loadFrame").fadeIn(300, onSuccess);
+	}
 
-	$('#items input[type="text"]:last').on('keydown', function(e) {
-	  if (e.keyCode == config.key("TAB")) {
-	      add_element();
-	  }
+	// Write localisable text to the page
+	string.fillinTextClasses();
+	$("#datasetname").prop("placeholder", lang("placeholder_datasetname"));
+	$("#datasetsubject").prop("title", lang("placeholder_subject"));
```

Comment:
Okay, but wouldn't it make life easier to just make sure that you only replace "placeholder"?